BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/IndoorPositioning.java

```diff
@@ -151,7 +155,11 @@ public static boolean unregisterLocationListener(LocationListener locationListen
 
             @Override
             public boolean matches(Beacon beacon) {
-                if (getInstance().indoorPositioningBeaconFilter != null && !getInstance().indoorPositioningBeaconFilter.matches(beacon)) {
+                BeaconFilter beaconFilter = getInstance().indoorPositioningBeaconFilter;
```

Comment:
This looks dangerous as this method is called when the instance is initialized but `indoorPositioningBeaconFilter` is initialized with null.

Reply:
Good catch, added null check